File: src/main/java/com/github/cao/awa/majalis/env/human/organ/heart/heartbeat/wave/p/PWaveTrigger.java
package com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.p;

import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveMetadata;
import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveTrigger;

/**
 * The trigger that engine the heartbeat wave P.
 */
public class PWaveTrigger extends WaveTrigger<PWaveMetadata, WaveMetadata> {
    @Override
    public PWaveMetadata primeMetadata() {
        return new PWaveMetadata();
    }
}

File: src/main/java/com/github/cao/awa/majalis/env/human/organ/heart/heartbeat/wave/pr/p/PWaveTrigger.kt
package com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.pr.p;

import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveMetadata;
import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveTrigger;
import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.pr.PRWaveMetadata;

/**
 * The trigger that engine the heartbeat wave P.
 */
public class PWaveTrigger extends WaveTrigger<PWaveMetadata, PRWaveMetadata> {
    @Override
    public PWaveMetadata primeMetadata() {
        return new PWaveMetadata();
    }
}
